### PR TITLE
Fixes for #2045 #2043 #2066 & other improvements 

### PR DESCRIPTION
**Pull request by dushansilva**
Fixes wso2/product-microgateway#2045 & wso2/product-microgateway#2043
Fixes wso2/product-microgateway#2066
& fixed same issue in userinfo, authorize and api key endpoints

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal [hunk 1]
```diff
@@ -279,6 +279,9 @@ function isAPILevelThrottled(string apiContext, string? apiVersion, ConditionalT
     boolean stopOnQuota;
     string apiThrottleKey = apiContext;
     if (apiVersion is string) {
+        if (!apiThrottleKey.endsWith("/" + apiVersion)) {
+            apiThrottleKey += "/" + apiVersion;
+        }
         apiThrottleKey += ":" + apiVersion;
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal:285]:
```suggestion
        if (!apiThrottleKey.endsWith('/' + apiVersion)) {
            apiThrottleKey += '/' + apiVersion;
        }
        apiThrottleKey += ':' + apiVersion;
```

File: components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal [hunk 2]
```diff
@@ -469,6 +472,9 @@ function setThrottleKeysWithVersion(RequestStreamDTO requestStreamDTO, http:Filt
         requestStreamDTO.apiVersion = apiVersion;
     }
     if (apiVersion is string) {
+        if (!requestStreamDTO.apiKey.endsWith("/" + apiVersion)) {
+            requestStreamDTO.apiKey += "/" + apiVersion;
+        }        
         requestStreamDTO.apiKey += ":" + apiVersion;
         requestStreamDTO.subscriptionKey += ":" + apiVersion;
```
Comment (praminda) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal:479]:
```suggestion
        if (!requestStreamDTO.apiKey.endsWith('/' + apiVersion)) {
            requestStreamDTO.apiKey += '/' + apiVersion;
        }        
        requestStreamDTO.apiKey += ':' + apiVersion;
        requestStreamDTO.subscriptionKey += ':' + apiVersion;
```

File: components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal [hunk 3]
```diff
@@ -279,7 +279,10 @@ function isAPILevelThrottled(string apiContext, string? apiVersion, ConditionalT
     boolean stopOnQuota;
     string apiThrottleKey = apiContext;
     if (apiVersion is string) {
-        apiThrottleKey += ":" + apiVersion;
+        if (!apiThrottleKey.endsWith('/' + apiVersion)) {
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal:282]: Do we test with and without conditional throttling polices

File: components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal [hunk 4]
```diff
@@ -469,8 +472,11 @@ function setThrottleKeysWithVersion(RequestStreamDTO requestStreamDTO, http:Filt
         requestStreamDTO.apiVersion = apiVersion;
     }
     if (apiVersion is string) {
-        requestStreamDTO.apiKey += ":" + apiVersion;
-        requestStreamDTO.subscriptionKey += ":" + apiVersion;
+        if (!requestStreamDTO.apiKey.endsWith('/' + apiVersion)) {
+            requestStreamDTO.apiKey += '/' + apiVersion;
```
[the remark this answers is not in the document]
Comment (dushansilva) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal]: tested and ok

File: components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal (hunk 3 above)
Comment (dushansilva) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/filters/throttle_filter.bal]: tested and ok

**Review by praminda — APPROVED** [review]

**Pull request merged**